cumulus/pallets/executor-registry/src/lib.rs

```diff
@@ -629,6 +736,34 @@ mod pallet {
     /// Total number of active executors.
     #[pallet::storage]
     pub(super) type TotalActiveExecutors<T> = StorageValue<_, u32, ValueQuery>;
+
+    /// Pending executor public key for the next epoch.
+    #[pallet::storage]
+    pub(super) type NextKey<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, ExecutorId, OptionQuery>;
+
+    /// A map tracking the owner of current and next key of each executor.
+    #[pallet::storage]
+    pub(super) type KeyOwner<T: Config> =
```

Comment:
I think there are still 2 edge-cases left:
* not inserted for executors in genesis config
* not removed on de-registration